packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -5,8 +5,8 @@ import * as util from '../util';
 
 type Options = [
   {
-    checkCompoundAssignments?: boolean;
     allowAny?: boolean;
+    skipCompoundAssignments?: boolean;
```

Comment:
I don't know if renaming it is overall super high value given the disruptive nature of the change. We do have precedence for "truthy-by-default" options like [the options in `strict-boolean-expressions`](https://typescript-eslint.io/rules/strict-boolean-expressions/#options).

I'll leave it up to you.